pkg/service/providerindex/legacy.go

```diff
@@ -55,16 +59,29 @@ func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStor
 	}
 
 	return LegacyClaimsStore{
-		contentToClaims: contentToClaimsMapper,
+		contentToClaims: contentToClaimsMappers,
 		claimsStore:     claimStore,
 		claimsAddr:      claimsAddr,
 	}, nil
 }
 
 // Find looks for the corresponding claims for a given content hash in the mapper and then fetches the claims from the
 // claims store
-func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Multihash) ([]model.ProviderResult, error) {
-	claimsCids, err := ls.contentToClaims.GetClaims(ctx, contentHash)
+func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Multihash, targetClaims []multicodec.Code) ([]model.ProviderResult, error) {
+	for _, mapper := range ls.contentToClaims {
+		claims, err := ls.findOne(ctx, contentHash, targetClaims, mapper)
+		if err != nil {
+			return nil, err
+		}
+		if len(claims) > 0 {
+			return claims, nil
+		}
+	}
```

Comment:
the results of this logic will depend on the order the mappers are provided in the slice. The results being returned will be the ones given by the first mapper returning any result.

Not saying it's wrong, but it may be worth stating it clearly somewhere (in the comment for `NewLegacyClaimsStore`, maybe?).

Reply:
yes that's the goal, but it's absolutely worst stating